clang/lib/Sema/SemaSYCL.cpp

```diff
@@ -1613,6 +1613,20 @@ class SyclKernelDecompMarker : public SyclKernelFieldHandler {
   }
 };
 
+static bool isSyclXilinxDDRBankType(const QualType &Ty) {
```

Comment:
Is it possible to be more generic, such as with ` static bool isSyclXilinxType(const QualType &Ty) {` wher it can introspect the types and guess what to do from some carefully crafted types in the `.hpp`?